### DISCUSSION TITLE
Plugin vs Config

### DISCUSSION THREAD

**Discussion by borisdamevin** [stylelint-stylistic/stylelint-stylistic#48] [Q&A] [answered]
Hi,

I would like to understand the difference between `@stylistic/stylelint-plugin` and ` @stylistic/stylelint-config`.

If a take ` @stylistic/eslint-plugin` for exemple, only plugin exist and can be overridered. But here I more preplexe.

Thank you.

**Comment by MorevM** [comment, marked as answer]
Good day @borisdamevin 

`@stylistic/stylelint-plugin` provides a set of rules without configuring them, meaning you have to do it yourself after installation to get the effect.
`@stylistic/stylelint-config` provides a ready configuration of these rules, but does not provide the rules themselves **directly**.

In fact, all `@stylistic/stylelint-config` does is ship [this file](https://github.com/stylelint-stylistic/stylelint-config/blob/main/stylelint.config.js), the ability to use the rules from the plugin - kind of a side-effect, since the config has the plugin as a dependency.

If you are familiar with the ESLint ecosystem, you may think of this as a "preset placed in a separate package for some reason".
So, if you prefer to configure all the style rules yourself - use the plugin, if you want to use (and show everyone else who looks in the `extends` section of your configuration) that you rely on the rules config that used to be in `stylelint-config-recommended` (I could be wrong, maybe those rules were in `stylelint-config-standard`, lazy to check) - then use the config package as your dependency.

---

The question why it was done this way is better address to whoever originally did the work of porting these rules from core (because I personally think it would have been better to keep everything in one place :) ).

You can learn more about the motivation behind these decisions from these issues:
* #9 
* #27

**Comment by borisdamevin** [comment]
Ok than you, not sure to all understand. @MorevM 

And the fact he `--fix` work on config and not with plugin is linked to this?

**Reply by MorevM** [reply]
If your `.stylelintrc.js` looks like:

```js
export default {
  plugins: [
    '@stylistic/stylelint-plugin',
  ],
};
```

Then no, running Stylelint with the `--fix` flag won't do anything since you don't have any configured rules from the plugin. 

---

If you manually configure some of rules like:

```js
export default {
  plugins: [
    '@stylistic/stylelint-plugin',
  ],
  rules: {
    '@stylistic/indentation': 'tab',
  },
};
```

In this example Stylelint **will** fix indentation errors, but **only** them, since the only rule is configured.

---

If you use the `@stylistic/stylelint-config` package and therefore the `extends` field in the config:

```js
export default {
  extends: [
    '@stylistic/stylelint-config',
  ],
};
```

Then running Stylelint with the `--fix` flag **will** fix **all** stylistic warnings according to the config you are extending - [source](https://github.com/stylelint-stylistic/stylelint-config/blob/main/stylelint.config.js).

**Comment by borisdamevin** [comment]
I try the second config, and nothing are fixed sadly, only warning are displayed. @MorevM

**Reply by MorevM** [reply]
How do you run Stylelint? Perhaps provide a minimum repository for reproduction...
As I can see, this should work as expected (you can see result on the right side with the fixer applied)

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0AXAFjAtAIwDYTAGsACYAHQDtjjJ8AneYumKAbkoF8QAaEAMwCWuGADkAhgFs4iGAA9JAB2FIwAZ1U9wECoIDmCEOSrEyIJQFddAiqtOMA2pWrVTAAVUoAnrgEeBYAHoPbxgfChRMCysKUydiAF1uONM6c2FbBFI4lxB3Lx8-QOtYcLEUAW07ExAUMWxY4w5OTUgdAV0AMQg6CTKDACtVbU1YBQ1EIxzg4TCUKtNcMpgPUyTjN2nfcsDp0OsIqOt5kEWUZbmQZo4gA)
